### PR TITLE
Added architecture flag znver3 for AOCC compiler 3.0 and above

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
march and mtune flangs are added for znver3 when AOCC 3.0 or above compiler is used.

### CODE REVIEW

**Comment by tgamblin** [comment]
LGTM!  Thanks!  Any chance gcc and clang could also be added here?

File: cpu/microarchitectures.json [hunk 1]
```diff
@@ -1328,6 +1328,47 @@
...
+        "ssse3",
+        "sse4_1",
+        "sse4_2",
+        "abm",
+        "xsavec",
+        "xsaveopt",
+        "clflushopt",
+        "popcnt",
+        "clwb"
+      ],
```
Comment (alalazo) [inline cpu/microarchitectures.json:1361]: Can you also provide a sample `/proc/cpuinfo` under the `test/targets` directory?

**Review by alalazo — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution!

I see that `znver3` has the same features as `zen2`. Is that correct or does it need to be complemented with new features? Also, it would be good to have a sample `/proc/cpuinfo` for regression tests.

File: cpu/microarchitectures.json (hunk 1 above)
Comment (alalazo) [reply cpu/microarchitectures.json:1361]: > Is the file '/proc/cpuinfo' file mandatory for this pull request to be accepted?

I'd say highly recommended. Without that it's impossible for `archspec` or other bindings (if there will be in the future) to unit test the detection of the architecture. Dumping the information from a single processor in a file named `linux-<os>-znver3` is fine. There are other examples under `tests/targets`

> Are you looking for the 'flags' section or any other specific section of the 'proc/cpuinfo' file, please let us know what you are looking for in this file

Any feature in the flags section of `/proc/cpuinfo` that may influence a binary executable. For other architectures we usually get those from GCC man pages, for instance for haswell the interesting flags are:
```
haswell
               Intel Haswell CPU with 64-bit extensions, MOVBE, MMX, SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2, POPCNT, AVX, AVX2, AES, PCLMUL,
               FSGSBASE, RDRND, FMA, BMI, BMI2 and F16C instruction set support.
```

The list you have now is "wrong" in the sense that it is very likely incomplete. Having the same set of flags as `zen2` means that any `zen2` uarch would be detected as `zen3` if this PR gets merged as is. Let me know if this needs clarification.

Comment (GaneshPrasadMA) [reply cpu/microarchitectures.json:1361]: Thanks for the explanation, please note that I will get back to you on the same

Comment (GaneshPrasadMA) [reply cpu/microarchitectures.json]: The above review comments have been addressed, kindly request for your review

File: tests/targets/linux-ubuntu20.04-znver3 [hunk 2]
```diff
@@ -0,0 +1,896 @@
...
+bugs		: sysret_ss_attrs spectre_v1 spectre_v2 spec_store_bypass
+bogomips	: 6786.81
+TLB size	: 2560 4K pages
+clflush size	: 64
+cache_alignment	: 64
+address sizes	: 48 bits physical, 48 bits virtual
+power management: ts ttp tm hwpstate cpb eff_freq_ro [13] [14]
```
Comment (tgamblin) [inline tests/targets/linux-ubuntu20.04-znver3:27]: @GaneshPrasadMA: we should only need one of the /proc/cpu records (as they're all pretty much the same).  Can you cut out the parts for procs 1-31 and leave just 0?

**Review by tgamblin — CHANGES_REQUESTED** [review]

File: tests/targets/linux-ubuntu20.04-znver3 (hunk 2 above)
Comment (GaneshPrasadMA) [reply tests/targets/linux-ubuntu20.04-znver3:27]: Please note that I have addressed the above review comment, and request for further review.

**Comment by alalazo** [comment]
@GaneshPrasadMA Thanks for the changes. I'll try to submit a draft PR in archspec based on this branch. If tests pass on that PR this would be fine with me. Otherwise I'll report the issues.

File: tests/targets/linux-ubuntu20.04-znver3 (hunk 2 above)
Comment (tgamblin) [reply tests/targets/linux-ubuntu20.04-znver3:27]: Thanks Ganesh1

**Comment by GaneshPrasadMA** [comment]
Yes, please report issues if any so that we can try to fix them.
I sincerely seek your guidance on the same as I am handling this magnitude of change in archspec for the first time.

**Review by alalazo — CHANGES_REQUESTED** [review]
@GaneshPrasadMA There's one just typo I induced you into. Can you please rename `linux-ubuntu20.04-znver3` into `linux-ubuntu20.04-zen3`? This is needed for unit test discovery in archspec. After that is done I think this is ready to be merged.

**Comment by GaneshPrasadMA** [comment]
Thanks for pointing that out. It has been corrected and changes as per the review comment have been pushed.

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Thanks @GaneshPrasadMA !

**Pull request merged**